firedrake/dmplexinc.pxi

```diff
@@ -63,7 +65,7 @@ cdef extern from "petscsf.h":
     int PetscSFReduceBegin(PETSc.PetscSF,MPI.MPI_Datatype,const void*, void*,MPI.MPI_Op)
     int PetscSFReduceEnd(PETSc.PetscSF,MPI.MPI_Datatype,const void*, void*,MPI.MPI_Op)
 
-cdef extern from "petscbt.h":
+cdef extern from "petscbt.h" nogil:
```

Comment:
It means that this function guarantees it doesn't invoke the python interpreter.  Cython emits some "refnanny" pre-/post-ambles to functions that might grab the gil.  I would like the new `use adjacency` function not to do this, which requires annotating that function, and its callees as `nogil`.